voice/src/main/kotlin/udp/AudioFramePoller.kt

```diff
@@ -0,0 +1,95 @@
+@file:OptIn(KordVoice::class)
+
+package dev.kord.voice.udp
+
+import dev.kord.common.annotation.KordVoice
+import dev.kord.voice.AudioProvider
+import dev.kord.voice.FrameInterceptor
+import dev.kord.voice.FrameInterceptorContext
+import dev.kord.voice.FrameInterceptorContextBuilder
+import io.ktor.network.sockets.*
+import io.ktor.utils.io.core.*
+import kotlinx.coroutines.*
+import mu.KotlinLogging
+import kotlin.coroutines.CoroutineContext
+import kotlin.math.max
+import kotlin.properties.Delegates
+import kotlin.time.Duration
+import kotlin.time.TimeSource
+
+private val audioFramePollerLogger = KotlinLogging.logger { }
+
+internal data class AudioFramePollerConfiguration(
+    val udp: DiscordUdpConnection,
+    val ssrc: Int,
+    val key: List<Byte>,
+    val provider: AudioProvider,
+    val baseFrameInterceptorContext: FrameInterceptorContextBuilder,
+    val interceptorFactory: (FrameInterceptorContext) -> FrameInterceptor
+)
+
+internal class AudioFramePollerConfigurationBuilder {
+    var udp: DiscordUdpConnection? = null
+    var ssrc: Int by Delegates.notNull()
+    var key: List<Byte> by Delegates.notNull()
+    var provider: AudioProvider by Delegates.notNull()
+    var baseFrameInterceptorContext: FrameInterceptorContextBuilder by Delegates.notNull()
+    var interceptorFactory: (FrameInterceptorContext) -> FrameInterceptor by Delegates.notNull()
```

Comment:
Make those part of the constructor

Reply:
Hard to do... This builder helps facilitate an easy way to build out this configuration over the time. As we won't ever have all this information all at once. They are split up into 3 different places, two different event listeners build them in UdpLifeCycleHandler, and finally in VoiceConnection with the method AudioFramePollerConfigurationBuilder#withConnection.

Reply:
so the builder is built over time?

Reply:
exactly